calcite-qml/demo/qml/main.qml

```diff
@@ -0,0 +1,224 @@
+/*******************************************************************************
+ *  Copyright 2012-2021 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import Esri.ArcGISRuntime 100.12
+import Calcite 1.0 as C
+import QtQuick.Layouts 1.12
+
+ApplicationWindow {
+    id: appWindow
+    width: 800
+    height: 600
+    title: "Calite_test"
+    visible: true
+
+    Component.onCompleted: {
+        //Calcite.theme = Calcite.Dark;
+    }
+
+    // add a mapView component
+    MapView {
+        id: mapView
+        anchors.fill: parent
+        // set focus to enable keyboard navigation
+        focus: true
+
+        // add a map to the mapview
+        Map {
+            // add the ArcGISStreets basemap to the map
+            BasemapImagery { }
```

Comment:
❌  The comment says streets, but code uses Imagery. Also, we should use the BasemapStyles

Reply:
Good catch. 

Could we have a quick discussion use of the basemap. I had deliberately chosen the older style basemaps as I didn't want to have the user set an API key to test the themes out in this demo application.

If we use the new-style basemaps - I would almost be tempted to strip out the basemaps entirely and make this a pure-Qt app without a runtime dependency for ease of use.

Reply:
The old basemap APIs will be deprecated soon, so I'd prefer to use it. You could use OSM Layer, which is not going away:

```
Map {
    Basemap {
        OpenStreetMapLayer {
        }
    }
}
```

Reply:
Yeah, it's good practice. It's a raster basemap, which isn't as smooth as the vector basemaps, but there's nothing wrong with it.